src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,151 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_START_CAPACITY = 16;
+    private static final int INCREASE_CAPACITY_VALUE = 2;
+    private int capacity;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] nodes;
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_START_CAPACITY;
+        this.size = 0;
+        this.threshold = (int) (DEFAULT_LOAD_FACTOR * capacity);
+        this.nodes = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resizeList();
+        }
+        int hashcode = hashCode(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        if (key == null && nodes[0] == null) {
+            nodes[0] = newNode;
+            size++;
+            return;
+        }
+        if (key == null) {
+            nodes[0] = newNode;
+            return;
+        }
+        int index = defineIndex(hashcode);
+        Node<K, V> lastNode = getLastCollisionNode(index);
+        if (lastNode == null) {
+            nodes[index] = newNode;
+            size++;
+        } else {
+            putNewNode(newNode, nodes[index]);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (key == null) {
+            return nodes[0].value;
+        }
+        int hashcode = hashCode(key);
+        int index = defineIndex(hashcode);
+        Node<K, V> firsNodeOnIndex = nodes[index];
+        if (firsNodeOnIndex == null) {
+            return null;
+        }
+        if (firsNodeOnIndex.next == null) {
+            return firsNodeOnIndex.value;
+        }
+        Node<K, V> foundNode = findNestedNode(firsNodeOnIndex, key);
+        return foundNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public int hashCode(K key) {
+        int hashcode = Objects.hash(key);
+        return hashcode < 0 ? (hashcode * -1) : hashcode;
+    }
+
+    private static class Node<K, V> {
```

Comment:
inner classes should be located at the end of the file

Reply:
++, changed